src/MBBSEmu.Btrieve/BtrieveFile.cs

```diff
@@ -214,77 +230,106 @@ public void LoadFile(ILogger logger, string path, string fileName)
 #endif
-            DeletedRecordOffsets = GetRecordPointerList(GetRecordPointer(0x10));
+            if (v6)
+            {
+                LoadPAT(logger);
+            }
+            else
+            {
+                DeletedRecordOffsets = GetRecordPointerList(GetRecordPointer(0x10));
+
+                LoadACS(logger, v6, -1);
+            }
 
-            LoadACS(logger);
-            LoadBtrieveKeyDefinitions(logger);
+
+            LoadBtrieveKeyDefinitions(logger, v6);
 
             //Only load records if there are any present
             if (RecordCount > 0)
-                LoadBtrieveRecords(logger);
+                LoadBtrieveRecords(logger, v6, allowCorruptedRecords);
         }
 
         /// <summary>
         ///     Validates the Btrieve database being loaded
         /// </summary>
         /// <returns>True if valid. If false, the string is the error message.</returns>
-        private (bool isValid, string errorMessage) ValidateDatabase()
+        private (bool isValid, bool v6, string errorMessage) ValidateDatabase(ILogger logger)
         {
             if (Data.Length < 2)
-                return (false, $"Btrieve File Is Empty/Invalid Length ({Data.Length} bytes)");
-            if (Data[0] == 'F' && Data[1] == 'C')
-                return (false, $"Cannot import v6 Btrieve database {FileName} - only v5 databases are supported for now. Please contact your ISV for a downgraded database.");
+                return (false, false, $"Btrieve File Is Empty/Invalid Length ({Data.Length} bytes)");
+            bool v6 = (Data[0] == 'F' && Data[1] == 'C' && Data[2] == 0 && Data[3] == 0);
             if (Data[0] != 0 && Data[1] != 0 && Data[2] != 0 && Data[3] != 0)
-                return (false, $"Doesn't appear to be a v5 Btrieve database {FileName}");
+                return (false, false, $"Doesn't appear to be a v5 Btrieve database {FileName}");
+
+            logger?.LogInformation(v6 ? "v6" : "v5");
 
-            var versionCode = Data[6] << 16 | Data[7];
-            switch (versionCode)
+            if (PageLength < 512 || (PageLength & 0x1FF) != 0)
+                return (false, v6, $"Invalid PageLength, must be multiple of 512 {FileName}");
+
+            _fcr = new byte[PageLength];
+
+            if (v6)
             {
-                case 3:
-                case 4:
-                case 5:
-                    break;
-                default:
-                    return (false, $"Invalid version code [{versionCode}] in v5 Btrieve database {FileName}");
+                // check the usage count to find the active FCR
+                var usageCount1 = Data[4] | Data[5] << 8 | Data[6] << 16 | Data[7] << 24;
+                var usageCount2 = Data[PageLength + 4] | Data[PageLength + 5] << 8 | Data[PageLength + 6] << 16 | Data[PageLength + 7] << 24;
+                // get the FCR
+                if (usageCount1 > usageCount2)
+                    Data.AsSpan().Slice(0, PageLength).CopyTo(_fcr.AsSpan());
```

Comment:
Dealers choice on this one, but you can use a range indexer on these now vs. slice if it starts from 0

`Data.AsSpan()[..PageLength].CopyTo(_fcr.AsSpan());`